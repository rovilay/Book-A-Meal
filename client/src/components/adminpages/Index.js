import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import navData from '../../helpers/navData';
import { setNav } from '../../actions/navLinks';


class MenuPage extends Component {
  componentDidMount() {
    this.props.dispatch(setNav(navData.adminNavDefault));
  }
  render() {
    return (
      <div>
        admin Page!
      </div>
    );
  }
}

MenuPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(MenuPage);
