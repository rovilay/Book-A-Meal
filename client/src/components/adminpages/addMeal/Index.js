import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddMeal extends Component {
  render() {
    return (
      <div>
        addmeal page
      </div>
    );
  }
}

export default AddMeal;
