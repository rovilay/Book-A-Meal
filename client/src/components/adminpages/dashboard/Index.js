/*
  eslint class-methods-use-this:0,
  eslint no-alert: 0,
*/
/* eslint no-restricted-globals: 0 */
import arraySort from 'array-sort';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  Accordion,
} from 'react-accessible-accordion';

import navData from '../../../helpers/navData';
import filterify from '../../../helpers/filterify';
import { setNav } from '../../../actions/navLinksActions';
import {
  getTodayMenu, getAllMenus, postMenu,
  updateMenu, addMealToNewMenu, removeMealFromNewMenu,
  emptyNewMenu, emptyEditMenu, addMealInEditMenu,
  deleteMealInEditMenu, setMenuForEdit, getMenuMeals,
  deleteMealInMenu, deleteMenuMeal
} from '../../../actions/menuActions';
import { getMeals } from '../../../actions/mealActions';
import {
  setModal, deleteMealInEditModal,
  addMealInEditMenuModal
} from '../../../actions/modalActions';
import MenuTable from './MenuTable/MenuTable';
import SetMenuCard from './MenuCard/SetMenu';
import Filter from '../../common/Filter';
import setFilter from '../../../actions/filterActions';
import ModalComp from '../Modal/Index';

class AdminDashboard extends Component {
  constructor(props) {
    super(props);

    this.setNewMenuMeal = this.setNewMenuMeal.bind(this);
    this.submitNewMenu = this.submitNewMenu.bind(this);
    this.showMenuDetails = this.showMenuDetails.bind(this);
    this.hideModal = this.hideModal.bind(this);
    this.editMenu = this.editMenu.bind(this);
    this.deleteRow = this.deleteRow.bind(this);
    this.onSubmitUpdate = this.onSubmitUpdate.bind(this);
    this.unCheckAll = this.unCheckAll.bind(this);
    this.getMenus = this.getMenus.bind(this);
    this.handlePaginationClick = this.handlePaginationClick.bind(this);
  }

  componentDidMount() {
    this.props.setNav(navData.adminNavDefault);
    this.props.getMeals({});
    this.props.getAllMenus({});
    this.props.setFilter({ filter: 'all' });
    this.hideModal();
  }

  onSubmitUpdate(menuDate, meals) {
    this.props.updateMenu({ menuDate, meals });
  }

  setNewMenuMeal(mealId) {
    let checkbox;
    if (this.props.modal.isEdit) {
      checkbox = document.getElementById(`${mealId}-edit`);
    } else {
      checkbox = document.getElementById(mealId);
    }
    if (checkbox.checked === true) {
      this.props.addMealToNewMenu(mealId);
    } else if (checkbox.checked === false) {
      this.props.removeMealFromNewMenu(mealId);
    }
  }

  /**
   * gets menus
   * @param {number} limit pagination limit
   * @param {number} offset pagination offset
   * @param {string} menuUrl menu meals url to get meals for a menu
   */
  getMenus({ limit, offset, menuUrl }) {
    this.props.getAllMenus({
      limit,
      offset,
      menuUrl
    });
  }

  hideModal() {
    this.props.setModal({
      isOpen: false,
      isEdit: false,
      isInfo: false,
      close: true,
      contentLabel: '',
      content: {}
    });
  }

  showMenuDetails(menuDetail) {
    this.props.setModal({
      isOpen: true,
      isInfo: true,
      isEdit: false,
      close: false,
      contentLabel: 'Menu details',
      content: { menuDetails: [...menuDetail] }
    });
  }

  editMenu({ menuId, postOn, meals }) {
    getMeals({});
    this.props.setModal({
      isOpen: true,
      isInfo: false,
      isEdit: true,
      close: false,
      contentLabel: 'Add Meals',
      content: { menuId, postOn, meals }
    });
    this.props.setMenuForEdit(meals);
    this.props.emptyNewMenu();
  }

  deleteRow(id) {
    this.props.deleteMealInEditModal(id);
    this.props.deleteMealInEditMenu(id);
  }

  unCheckAll(mealIdArr) {
    mealIdArr.forEach((mealId) => {
      const checkbox = document.getElementById(mealId);
      checkbox.checked = false;
    });
  }

  submitNewMenu() {
    const meals = [...this.props.newMenuMeals];
    const date = document.getElementById('postOn').value;
    const postOn = date.split('/').reverse().join('-');
    this.props.postMenu({ postOn, meals });
  }


  /**
   * handles pagination changes
   *
   * @param {object} data data object from pagination component
   */
  handlePaginationClick(data) {
    // const nextPage = data.selected + 1;
    const { limit } = this.props.pagination;
    const offset = (data.selected) * limit;
    this.getMenus({ limit, offset });
  }

  render() {
    const { user } = this.props;
    const { firstName, lastName } = user;
    return (
      <div>
        <div className="welcome">
          <p className="merienda">
            welcome, {firstName} {lastName}
          </p>
        </div>
        <section className="setmenu">
          <Accordion>
            <div className="setmenu-container">
              <SetMenuCard
                setNewMenuMeal={this.setNewMenuMeal}
                submitNewMenu={this.submitNewMenu}
                {...this.props}
              />
            </div>
          </Accordion>
        </section>
        {/* <MenuAccordion /> */}
        <section className="adminpage">
          <div className="menu-title">Menus List</div>
          <div className="container">
            <Filter
              {...this.props}
              tableContent="Menus_List"
            />
            {
              (this.props.menus.length === 0)
                ?
                  <p className="empty not-found">No menu found!</p>
                :
                  <Accordion>
                    <MenuTable
                      showMenuDetails={this.showMenuDetails}
                      editMenu={this.editMenu}
                      {...this.props}
                    />
                  </Accordion>
            }
          </div>
        </section>
        <ModalComp
          hideModal={this.hideModal}
          deleteRow={this.deleteRow}
          submitUpdate={this.onSubmitUpdate}
          setNewMenuMeal={this.setNewMenuMeal}
          {...this.props}
        />
      </div>
    );
  }
}

AdminDashboard.propTypes = {
  setNav: PropTypes.func.isRequired,
  getMeals: PropTypes.func.isRequired,
  getAllMenus: PropTypes.func.isRequired,
  meals: PropTypes.array.isRequired,
  addMealToNewMenu: PropTypes.func.isRequired,
  removeMealFromNewMenu: PropTypes.func.isRequired,
  postMenu: PropTypes.func.isRequired,
  newMenuMeals: PropTypes.array.isRequired,
  setModal: PropTypes.func.isRequired,
  deleteMealInEditModal: PropTypes.func.isRequired,
  updateMenu: PropTypes.func.isRequired,
  setMenuForEdit: PropTypes.func.isRequired,
  emptyNewMenu: PropTypes.func.isRequired,
  modal: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
  menus: PropTypes.array.isRequired,
  setFilter: PropTypes.func.isRequired,
  deleteMealInEditMenu: PropTypes.func.isRequired,
  pagination: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  meals: arraySort(state.meal.meals, 'title'),
  newMenuMeals: state.menu.newMenu,
  menus: filterify(state.menu.allMenus, state.filter),
  menuMeals: state.menu.menuMeals,
  pagination: state.menu.pagination,
  menuMealsPagination: state.menu.menuMealsPagination,
  modal: state.modal,
  editMenuMeals: state.menu.editMenu,
  mealPagination: state.meal.pagination,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setNav,
    setModal,
    getMeals,
    emptyNewMenu,
    getAllMenus,
    getTodayMenu,
    updateMenu,
    postMenu,
    emptyEditMenu,
    addMealToNewMenu,
    removeMealFromNewMenu,
    setMenuForEdit,
    deleteMealInEditMenu,
    deleteMealInMenu,
    deleteMealInEditModal,
    addMealInEditMenu,
    addMealInEditMenuModal,
    getMenuMeals,
    setFilter,
    deleteMenuMeal
  },
  dispatch
);

export default connect(mapStateToProps, mapDispatchToProps)(AdminDashboard);
