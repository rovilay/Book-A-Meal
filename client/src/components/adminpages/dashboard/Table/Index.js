import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import tableHead from '../../../../helpers/tableHead';
import TableHead from '../../../common/Table/TableHead';
import MenuTableRow from './MenuTableRow';

class MenuTable extends Component {
  render() {
    const { menus } = this.props;
    return (
      <div className="menu_table">
        <table>
          <TableHead tableHead={tableHead.menuTableHead} />
          <tbody>
            {
              menus.map((menu, i) => {
                const {
                  id: menuId,
                  postOn,
                  User,
                } = menu;
                const item = {
                  sn: ++i,
                  menuId,
                  postOn: moment(postOn).format('LL'),
                  createdBy: `${User.firstName} ${User.lastName}`
                };
                return (<MenuTableRow
                  key={menuId}
                  item={item}
                />);
              })
            }
          </tbody>
        </table>
      </div>
    );
  }
}

MenuTable.propTypes = {
  menus: PropTypes.array.isRequired
};

export default MenuTable;
