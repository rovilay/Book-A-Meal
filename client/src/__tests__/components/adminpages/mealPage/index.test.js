import React from 'react';
import { shallow } from 'enzyme';
import { MealPage, mapStateToProps  } from '../../../../components/adminpages/mealPage/Index';
import { mealDefaultState } from '../../../../reducers/mealReducer';
import { filterDefaultState } from '../../../../reducers/filterReducer';
import { meals } from '../../../__mockData__/mealMock';

const props = {
  mealError: 'Error',
  setNav: jest.fn(),
  getMeals: jest.fn(),
  setMealForEdit: jest.fn(),
  removeMealFromEdit: jest.fn(),
  mealOnEdit: {
    ...meals[0],
    description: 'so sweet'
  },
  postMeal: jest.fn(),
  updateMeal: jest.fn(),
  deleteMeal: jest.fn(),
  setFilter: jest.fn(),
  meals,
  pagination: {
    limit: 12,
    count: 13,
    numOfPages: 2
  },
  updateMealOnEdit: jest.fn()
};

describe('MealPage component test', () => {
  const setup = () => shallow(<MealPage {...props} />);


  it('should render MealPage correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });


  it('should call `onUpdateMeal` if update meal button is clicked', (done) => {
    const wrapper = setup();

    const onUpdateMealSpy= jest.spyOn(wrapper.instance(), 'onUpdateMeal');
    wrapper.instance().onUpdateMeal(props.mealOnEdit.id)
    expect(onUpdateMealSpy).toHaveBeenCalled();

    done();
  });


  it('should call `getFormValues` when submitting meal form', (done) => {
    const wrapper = setup();
    wrapper.setState({
      isEdit: false
    })

    document.body.innerHTML =
    `<div>
      <input type="text" id="meal-name" value="Eba and Egusi" />
      <input type="text" id="dsc" value="So sweet" />
      <input type="number" id="price" value="200" />
    </div>`;

    const getFormValuesSpy= jest.spyOn(wrapper.instance(), 'getFormValues');
    wrapper.instance().getFormValues()
    expect(getFormValuesSpy).toHaveBeenCalled();

    done();
  });


  it('should call `handleChange` if meal form input is being changed', (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: 'title',
        value: 'Eba and Egusi'
      }
    }

    const handleChangeSpy= jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().handleChange(event)
    expect(handleChangeSpy).toHaveBeenCalled();

    done();
  });


  it('should call `checkFileSize` if meal image is being uploaded on meal form',
  (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    }

    document.body.innerHTML =
    `<div>
      <input type="file" id="meal-image" value="image.png" />
    </div>`;

    const checkFileSizeSpy= jest.spyOn(wrapper.instance(), 'checkFileSize');
    wrapper.instance().checkFileSize(event)
    expect(checkFileSizeSpy).toHaveBeenCalled();

    done();
  });


  it('should call `fillForm` on edit button click', (done) => {
    const wrapper = setup();

    document.body.innerHTML =
    `<div>
      <input type="text" id="meal-name" value="Eba and Egusi" />
      <input type="text" id="dsc" value="So sweet" />
      <input type="number" id="price" value="200" />
    </div>`;

    const fillFormSpy= jest.spyOn(wrapper.instance(), 'fillForm');
    wrapper.instance().fillForm()
    expect(fillFormSpy).toHaveBeenCalled();

    done();
  });


  it('should call `clearForm` on on back button click', (done) => {
    const wrapper = setup();

    document.body.innerHTML =
    `<div>
      <input type="text" id="meal-name" value="Eba and Egusi" />
      <input type="text" id="dsc" value="So sweet" />
      <input type="number" id="price" value="200" />
      <input type="file" id="meal-image" value="200" />
    </div>`;

    const clearFormSpy= jest.spyOn(wrapper.instance(), 'clearForm');
    wrapper.instance().clearForm()
    expect(clearFormSpy).toHaveBeenCalled();

    done();
  });


  it('should call `handlePaginationClick` if pagination button is clicked',
  (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(wrapper.instance(), 'handlePaginationClick');
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });


  it('should call `closeEdit` on update button click', (done) => {
    const wrapper = setup();

    const closeEditSpy= jest.spyOn(wrapper.instance(), 'closeEdit');
    wrapper.instance().closeEdit()
    expect(closeEditSpy).toHaveBeenCalled();

    done();
  });

  it('should call `editMeal` on edit button click', (done) => {
    const wrapper = setup();

    const editMealSpy= jest.spyOn(wrapper.instance(), 'editMeal');
    wrapper.instance().editMeal(meals[0].id)
    expect(editMealSpy).toHaveBeenCalled();

    done();
  });


  it('should call `onAddMeal` if add meal form is submitted', (done) => {
    const wrapper = setup();
    wrapper.setState({
      isEdit: false
    })

    document.body.innerHTML =
    `<div>
      <input type="text" id="meal-name" value="Eba and Egusi" />
      <input type="text" id="dsc" value="So sweet" />
      <input type="number" id="price" value="200" />
    </div>`;

    const onAddMealSpy= jest.spyOn(wrapper.instance(), 'onAddMeal');
    wrapper.instance().onAddMeal()
    expect(onAddMealSpy).toHaveBeenCalled();

    done();
  });


  it('should map state to props', () => {
    const initialState = {
      filter: {
        ...filterDefaultState
      },
      meal: {
        ...mealDefaultState
      }
    };

    const ownProps = {
      meals,
      mealOnEdit: {
        ...meals[0],
        description: 'so sweet'
      },
      mealError: 'Error',
      pagination: {
        limit: 12,
        offset: 0,
        count: 13,
        nextOffset: 12
      }
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  })
});
