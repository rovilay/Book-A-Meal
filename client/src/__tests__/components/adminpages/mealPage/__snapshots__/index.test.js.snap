// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealPage component test should map state to props 1`] = `
Object {
  "mealError": "",
  "mealOnEdit": Object {},
  "meals": Array [],
  "pagination": Object {
    "count": 0,
    "limit": 12,
    "numOfPages": 1,
    "offset": 0,
  },
}
`;

exports[`MealPage component test should render MealPage correctly 1`] = `
<div
  className="meal-page"
>
  <div
    className="welcome merienda"
  >
    <p
      className="merienda"
    >
      Manage Meals
    </p>
  </div>
  <section
    className="setMeal"
  >
    <AccordionWrapper
      accordion={true}
      className="accordion"
      onChange={[Function]}
    >
      <div
        className="setMeal-container"
      >
        <MealForm
          addMeal={[Function]}
          closeEdit={[Function]}
          deleteMeal={[MockFunction]}
          disableBtn={false}
          getMeals={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "limit": 12,
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          handleChange={[Function]}
          imageToUpload=""
          isEdit={false}
          mealError="Error"
          mealOnEdit={
            Object {
              "description": "so sweet",
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            }
          }
          meals={
            Array [
              Object {
                "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
                "portion": 2,
                "price": 1000,
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
                "portion": 3,
                "price": 400,
                "title": "SALAD",
              },
              Object {
                "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
                "portion": 1,
                "price": 600,
                "title": "SPAGHETTI",
              },
              Object {
                "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
                "portion": 2,
                "price": 500,
                "title": "YAM AND EGG",
              },
            ]
          }
          pagination={
            Object {
              "count": 13,
              "limit": 12,
              "numOfPages": 2,
            }
          }
          postMeal={[MockFunction]}
          removeMealFromEdit={[MockFunction]}
          setFilter={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "filter": "all",
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setMealForEdit={[MockFunction]}
          setNav={
            [MockFunction] {
              "calls": Array [
                Array [
                  Array [
                    Object {
                      "link": "/dashboard",
                      "title": "Manage Menus",
                    },
                    Object {
                      "link": "/mealpage",
                      "title": "Manage Meals",
                    },
                    Object {
                      "link": "/orderHistory",
                      "title": "Orders",
                    },
                    Object {
                      "link": "/",
                      "title": "Log Out",
                    },
                  ],
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          showUploadBar={[Function]}
          updateMeal={[Function]}
          updateMealOnEdit={[MockFunction]}
          uploadImage={[Function]}
          uploadedImageLink=""
        />
      </div>
    </AccordionWrapper>
  </section>
  <div
    className=""
  >
    <div
      className="title"
    >
      Meals
    </div>
    <FilterComp
      deleteMeal={[MockFunction]}
      getMeals={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "limit": 12,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      mealError="Error"
      mealOnEdit={
        Object {
          "description": "so sweet",
          "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
          "portion": 2,
          "price": 1000,
          "title": "EBA AND EGUSI",
        }
      }
      meals={
        Array [
          Object {
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          },
          Object {
            "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
            "portion": 3,
            "price": 400,
            "title": "SALAD",
          },
          Object {
            "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
            "portion": 1,
            "price": 600,
            "title": "SPAGHETTI",
          },
          Object {
            "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
            "portion": 2,
            "price": 500,
            "title": "YAM AND EGG",
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      postMeal={[MockFunction]}
      removeMealFromEdit={[MockFunction]}
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setMealForEdit={[MockFunction]}
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      tableContent="caterer_meals"
      updateMeal={[MockFunction]}
      updateMealOnEdit={[MockFunction]}
    />
    <div
      className="menu-container"
    >
      <MealCard
        deleteMeal={[MockFunction]}
        editMeal={[Function]}
        getMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "limit": 12,
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        key="6edf5100-3313-4a4f-8a9d-a1bf79b0a72f"
        mealData={
          Object {
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          }
        }
        mealError="Error"
        mealOnEdit={
          Object {
            "description": "so sweet",
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          }
        }
        meals={
          Array [
            Object {
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
              "portion": 3,
              "price": 400,
              "title": "SALAD",
            },
            Object {
              "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
              "portion": 1,
              "price": 600,
              "title": "SPAGHETTI",
            },
            Object {
              "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
              "portion": 2,
              "price": 500,
              "title": "YAM AND EGG",
            },
          ]
        }
        pagination={
          Object {
            "count": 13,
            "limit": 12,
            "numOfPages": 2,
          }
        }
        postMeal={[MockFunction]}
        removeMealFromEdit={[MockFunction]}
        setFilter={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "filter": "all",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setMealForEdit={[MockFunction]}
        setNav={
          [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Object {
                    "link": "/dashboard",
                    "title": "Manage Menus",
                  },
                  Object {
                    "link": "/mealpage",
                    "title": "Manage Meals",
                  },
                  Object {
                    "link": "/orderHistory",
                    "title": "Orders",
                  },
                  Object {
                    "link": "/",
                    "title": "Log Out",
                  },
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        updateMeal={[MockFunction]}
        updateMealOnEdit={[MockFunction]}
      />
      <MealCard
        deleteMeal={[MockFunction]}
        editMeal={[Function]}
        getMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "limit": 12,
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        key="ce41fb68-4a51-4033-8c3a-e5a27eadcc91"
        mealData={
          Object {
            "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
            "portion": 3,
            "price": 400,
            "title": "SALAD",
          }
        }
        mealError="Error"
        mealOnEdit={
          Object {
            "description": "so sweet",
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          }
        }
        meals={
          Array [
            Object {
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
              "portion": 3,
              "price": 400,
              "title": "SALAD",
            },
            Object {
              "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
              "portion": 1,
              "price": 600,
              "title": "SPAGHETTI",
            },
            Object {
              "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
              "portion": 2,
              "price": 500,
              "title": "YAM AND EGG",
            },
          ]
        }
        pagination={
          Object {
            "count": 13,
            "limit": 12,
            "numOfPages": 2,
          }
        }
        postMeal={[MockFunction]}
        removeMealFromEdit={[MockFunction]}
        setFilter={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "filter": "all",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setMealForEdit={[MockFunction]}
        setNav={
          [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Object {
                    "link": "/dashboard",
                    "title": "Manage Menus",
                  },
                  Object {
                    "link": "/mealpage",
                    "title": "Manage Meals",
                  },
                  Object {
                    "link": "/orderHistory",
                    "title": "Orders",
                  },
                  Object {
                    "link": "/",
                    "title": "Log Out",
                  },
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        updateMeal={[MockFunction]}
        updateMealOnEdit={[MockFunction]}
      />
      <MealCard
        deleteMeal={[MockFunction]}
        editMeal={[Function]}
        getMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "limit": 12,
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        key="58634b35-2254-40de-a5eb-3c1166ac56c0"
        mealData={
          Object {
            "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
            "portion": 1,
            "price": 600,
            "title": "SPAGHETTI",
          }
        }
        mealError="Error"
        mealOnEdit={
          Object {
            "description": "so sweet",
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          }
        }
        meals={
          Array [
            Object {
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
              "portion": 3,
              "price": 400,
              "title": "SALAD",
            },
            Object {
              "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
              "portion": 1,
              "price": 600,
              "title": "SPAGHETTI",
            },
            Object {
              "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
              "portion": 2,
              "price": 500,
              "title": "YAM AND EGG",
            },
          ]
        }
        pagination={
          Object {
            "count": 13,
            "limit": 12,
            "numOfPages": 2,
          }
        }
        postMeal={[MockFunction]}
        removeMealFromEdit={[MockFunction]}
        setFilter={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "filter": "all",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setMealForEdit={[MockFunction]}
        setNav={
          [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Object {
                    "link": "/dashboard",
                    "title": "Manage Menus",
                  },
                  Object {
                    "link": "/mealpage",
                    "title": "Manage Meals",
                  },
                  Object {
                    "link": "/orderHistory",
                    "title": "Orders",
                  },
                  Object {
                    "link": "/",
                    "title": "Log Out",
                  },
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        updateMeal={[MockFunction]}
        updateMealOnEdit={[MockFunction]}
      />
      <MealCard
        deleteMeal={[MockFunction]}
        editMeal={[Function]}
        getMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "limit": 12,
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        key="959f0675-0ad3-47b7-8401-ec9e10bc1863"
        mealData={
          Object {
            "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
            "portion": 2,
            "price": 500,
            "title": "YAM AND EGG",
          }
        }
        mealError="Error"
        mealOnEdit={
          Object {
            "description": "so sweet",
            "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
            "portion": 2,
            "price": 1000,
            "title": "EBA AND EGUSI",
          }
        }
        meals={
          Array [
            Object {
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
              "portion": 3,
              "price": 400,
              "title": "SALAD",
            },
            Object {
              "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
              "portion": 1,
              "price": 600,
              "title": "SPAGHETTI",
            },
            Object {
              "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
              "portion": 2,
              "price": 500,
              "title": "YAM AND EGG",
            },
          ]
        }
        pagination={
          Object {
            "count": 13,
            "limit": 12,
            "numOfPages": 2,
          }
        }
        postMeal={[MockFunction]}
        removeMealFromEdit={[MockFunction]}
        setFilter={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "filter": "all",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setMealForEdit={[MockFunction]}
        setNav={
          [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Object {
                    "link": "/dashboard",
                    "title": "Manage Menus",
                  },
                  Object {
                    "link": "/mealpage",
                    "title": "Manage Meals",
                  },
                  Object {
                    "link": "/orderHistory",
                    "title": "Orders",
                  },
                  Object {
                    "link": "/",
                    "title": "Log Out",
                  },
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        updateMeal={[MockFunction]}
        updateMealOnEdit={[MockFunction]}
      />
    </div>
    <div
      className="pagination-container"
    >
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <a
            href=""
          >
            ...
          </a>
        }
        containerClassName="pagination"
        disableInitialCallback={false}
        disabledClassName="disabled"
        marginPagesDisplayed={2}
        nextClassName="next"
        nextLabel=">>"
        onPageChange={[Function]}
        pageCount={2}
        pageRangeDisplayed={5}
        previousClassName="previous"
        previousLabel="<<"
        subContainerClassName="pages pagination"
      />
    </div>
  </div>
</div>
`;
